tiledb/sm/rest/curl.cc

```diff
@@ -180,21 +181,80 @@ static int buffer_list_seek_callback(
   return CURL_SEEKFUNC_FAIL;
 }
 
+/**
+ * This callback function gets called by libcurl as soon as a header has been
+ * received. libcurl buffers headers and delivers only "full" headers, one by
+ * one, to this callback. This callback should return the number of bytes
+ * actually taken care of if that number differs from the number passed to
+ * your callback function, it signals an error condition to the library
+ *
+ * @param res_data points to the delivered data
+ * @param size the size of that data is size multiplied with nmemb
+ * @param count number of bytes actually taken care of
+ * @param userdata the userdata argument is set with CURLOPT_HEADERDATA
+ * @return
+ */
+size_t write_header_callback(
+    void* res_data, size_t size, size_t count, void* userdata) {
+  size_t header_length = size * count;
+  auto* header_buffer = static_cast<char*>(res_data);
+  auto* pmHeader = static_cast<HeaderCbData*>(userdata);
```

Comment:
If `rest_components_from_url()` failed, it would return an empty string. In that scenario, `pmHeader->uri` will be empty.

We expect that `pmHeader->uri` will always be non-empty, so we should immediately check:
```
if (pmHeader->uri.empty()) {
  LOG_STATUS("...");
  return 0;
}
```

This is to protect us